### ISSUE TITLE
Command line invocation of macros with parameters containing a parenthesis will fail if Imagej is already open

### ISSUE THREAD

**Issue by Sethur** [imagej/imagej#164]
The issue is related to the following earlier issue:

[https://list.nih.gov/cgi-bin/wa.exe?A2=ind1512&L=IMAGEJ&P=R29972&1=IMAGEJ&9=A&J=on&d=No+Match%3BMatch%3BMatches&z=4](https://list.nih.gov/cgi-bin/wa.exe?A2=ind1512&L=IMAGEJ&P=R29972&1=IMAGEJ&9=A&J=on&d=No+Match%3BMatch%3BMatches&z=4)

This was fixed, thus now something like:

`imagej-win64.exe -macro "mymacro.ijm" "tmp[A(.dcm"`

will not result in an error on the first invocation.

However, if one leaves the **instance of ImageJ running** and repeats this process, there will still be an "Macro or script not found" error, with the relevant part of the debug log showing:

[DEBUG] SocketServer.sendArgument: "user.dir C:\Programs\ImageJ"
[DEBUG] SocketServer.sendArgument: "macro ImportDicom.ijm(tmp[A()"

After this failure, a second instance of ImageJ is opened handling the macro execution correctly.

I assume this happens because the way macros are executed when there is already a running instance of ImageJ is different. 

This issue is very annoying when dealing with e.g. DICOM files containing square brackets and parentheses since it effectively prevents the user from opening handling multiple datasets effectively as starting a new ImageJ instance is quite slow at the moment.

**Comment by ctrueden** [comment]
@Sethur Did you try going to _Edit &#9654; Options &#9654; Misc_ and unchecking the "Run single instance listener" option?

**Comment by Sethur** [comment]
@ctrueden Yes, I meant that the arguments are not passed correctly to the running instance. Further fiddling meanwhile revealed that the issue **only** occurs with ImageJ2. It can be avoided using `--ij1` with the launcher, thus disabling ImageJ2 support. Apparently, there is a bug in the way the instance listener  works when IJ2 support is enabled. A workaround for me now is to mask the parentheses, i.e. `(` and `)` when passing the path argument to my DICOM import macro, and then substitute them again in the macro, like so:

In the batch file that launches ImageJ2:
--------------------------------------------
```
set FILENAME=%ARG1:(=___----%
set FILENAME=%FILENAME:)=____---%

imagej-win64 --no-splash -macro "ImportDICOM.ijm" %FILENAME%
```

In the macro:
---------------
```
fileName = getArgument;
if (fileName=="") exit ("No file name given!");

// Workaround for bug in ImageJ where parentheses in filenames lead to errors on multiple CLI invocations
fileName = replace(fileName, "___----", "(");
fileName = replace(fileName, "____---", ")");

// Open parent folder
if (File.isDirectory(fileName))
    open(fileName);
else
    open(File.getParent(fileName));
```

**Comment by ctrueden** [comment]
@Sethur Glad you have a workaround.

The single instance listener code indeed works differently in ImageJ2. It was actually unsupported for a long time, but then finally readded as a project sponsored by Zeiss. Unfortunately, the developers who have worked on it in the past have left the project, and I do not personally have a good understanding of how it works. If you feel strongly about fixing this, you can [have a look at the code here](https://github.com/imagej/imagej-legacy/blob/imagej-legacy-0.23.5/src/main/java/net/imagej/legacy/SingleInstance.java#L45-L56). Otherwise, this bug will likely not be fixed any time soon, due to lack of resources.
